src/miner.cpp

```diff
@@ -351,332 +385,387 @@ CBlockTemplate* CreateNewBlock(const CChainParams& chainparams, const MinerAddre
-
-        int64_t nLockTimeCutoff = (STANDARD_LOCKTIME_VERIFY_FLAGS & LOCKTIME_MEDIAN_TIME_PAST)
-                                ? nMedianTimePast
-                                : pblock->GetBlockTime();
-
-        bool fPriorityBlock = nBlockPrioritySize > 0;
-        if (nonEmptyBlock && fPriorityBlock) {
-            vecPriority.reserve(mempool.mapTx.size());
-            for (CTxMemPool::indexed_transaction_set::iterator mi = mempool.mapTx.begin();
-                 mi != mempool.mapTx.end(); ++mi)
-            {
-                double dPriority = mi->GetPriority(nHeight);
-                CAmount dummy;
-                mempool.ApplyDeltas(mi->GetTx().GetHash(), dPriority, dummy);
-                vecPriority.push_back(TxCoinAgePriority(dPriority, mi));
-            }
-            std::make_heap(vecPriority.begin(), vecPriority.end(), pricomparer);
+        if (!inBlock.count(parent)) {
+            return true;
         }
+    }
+    return false;
+}
 
-        CTxMemPool::indexed_transaction_set::index<mining_score>::type::iterator mi = mempool.mapTx.get<mining_score>().begin();
-        CTxMemPool::txiter iter;
-
-        // We want to track the value pool, but if the miner gets
-        // invoked on an old block before the hardcoded fallback
-        // is active we don't want to trip up any assertions. So,
-        // we only adhere to the turnstile (as a miner) if we
-        // actually have all of the information necessary to do
-        // so.
-        CAmount sproutValue = 0;
-        CAmount saplingValue = 0;
-        CAmount orchardValue = 0;
-        bool monitoring_pool_balances = true;
-        if (chainparams.ZIP209Enabled()) {
-            if (pindexPrev->nChainSproutValue) {
-                sproutValue = *pindexPrev->nChainSproutValue;
-            } else {
-                monitoring_pool_balances = false;
-            }
-            if (pindexPrev->nChainSaplingValue) {
-                saplingValue = *pindexPrev->nChainSaplingValue;
-            } else {
-                monitoring_pool_balances = false;
-            }
-            if (pindexPrev->nChainOrchardValue) {
-                orchardValue = *pindexPrev->nChainOrchardValue;
-            } else {
-                monitoring_pool_balances = false;
-            }
+
+
+bool BlockAssembler::TestForBlock(CTxMemPool::txiter iter)
+{
+    if (nBlockSize + iter->GetTxSize() >= nBlockMaxSize) {
+        // If the block is so close to full that no more txs will fit
+        // or if we've tried more than 50 times to fill remaining space
+        // then flag that the block is finished
+        if (nBlockSize >  nBlockMaxSize - 100 || lastFewTxs > 50) {
+             blockFinished = true;
+             return false;
+        }
+        // Once we're within 1000 bytes of a full block, only look at 50 more txs
+        // to try to fill the remaining space.
+        if (nBlockSize > nBlockMaxSize - 1000) {
```

Comment:
This is the same `1000` constant as was commented on above; we should probably create a constant for it.